nifi-extension-bundles/nifi-standard-bundle/nifi-standard-processors/src/test/java/org/apache/nifi/processors/standard/TestValidateRecord.java

```diff
@@ -593,6 +599,47 @@ public void testValidateMaps() throws IOException, InitializationException, Malf
         }
     }
 
+    @ParameterizedTest
+    @MethodSource("getLocales")
+    void testTimeZoneWithTimeStamp(Locale locale) throws Exception {
+        Locale originalLocale = Locale.getDefault();
+        Locale.setDefault(locale);
+        final String timestampWithTimeZonePattern = "dd/MM/yyyy HH:mm:ssZ";
+        final String schema = new String(Files.readAllBytes(Paths.get("src/test/resources/TestValidateRecord/timestampWithTimeZone.avsc")), "UTF-8");
+        final CSVReader csvReader = new CSVReader();
+        runner.addControllerService("reader", csvReader);
+        runner.setProperty(csvReader, SchemaAccessUtils.SCHEMA_ACCESS_STRATEGY, SchemaAccessUtils.SCHEMA_TEXT_PROPERTY);
+        runner.setProperty(csvReader, SchemaAccessUtils.SCHEMA_TEXT, schema);
+        runner.setProperty(csvReader, CSVUtils.FIRST_LINE_IS_HEADER, "true");
+        runner.setProperty(csvReader, CSVUtils.VALUE_SEPARATOR, "◆");
+        runner.setProperty(csvReader, DateTimeUtils.TIMESTAMP_FORMAT, timestampWithTimeZonePattern);
+        runner.enableControllerService(csvReader);
+
+        final CSVRecordSetWriter csvWriter = new CSVRecordSetWriter();
+        runner.addControllerService("writer", csvWriter);
+        runner.setProperty(csvWriter, "Schema Write Strategy", "full-schema-attribute");
+        runner.setProperty(csvWriter, DateTimeUtils.TIMESTAMP_FORMAT, timestampWithTimeZonePattern);
+        runner.enableControllerService(csvWriter);
+
+        runner.setProperty(ValidateRecord.RECORD_READER, "reader");
+        runner.setProperty(ValidateRecord.RECORD_WRITER, "writer");
+        runner.setProperty(ValidateRecord.ALLOW_EXTRA_FIELDS, "false");
+        runner.setProperty(ValidateRecord.MAX_VALIDATION_DETAILS_LENGTH, "4000");
+        runner.setProperty(ValidateRecord.VALIDATION_DETAILS_ATTRIBUTE_NAME, "valDetails");
+        final String timestampWithTimezone = "24/09/2024 15:04:23+0630";
+        final String content = "apache_date◆apache_ip_source◆apache_method◆apache_path◆apache_query_string◆apache_response_code◆apache_referer◆apache_user_agent\n" +
+                timestampWithTimezone + "◆10.4.3.20◆GET◆/path◆?test=toto◆200◆-◆";
+
+        runner.enqueue(content);
+        runner.run();
+
+        runner.assertTransferCount(ValidateRecord.REL_VALID, 1);
+        final MockFlowFile validFlowFile = runner.getFlowFilesForRelationship(ValidateRecord.REL_VALID).getFirst();
+        //Validate timezone is taken into account
+        assertFalse(validFlowFile.getContent().contains(timestampWithTimezone));
```

Comment:
The comment does not seem to align with the assertion. This seems to indicate that the timestamp is _not_ expected to be found?

Reply:
I see how its not clear but ultimately I want to assert that the `15` in `15:04:23` is no longer there which should demonstrate the timezone is taken into account.

Reply:
In other words, the test is expecting the timestamp to change?

Reply:
Yes

Reply:
Thanks for confirming. That highlights the core concern. The current behavior of the `ObjectTimestampFieldConverter` expects the input and output values to be the same, which aligns with the understanding the `java.sql.Timestamp` does not have a zone offset. When converting from a "zoned" timestamp as in this test, it raises the question whether the output should reflect that offset when computing output timestamp.

Perhaps one way to make the test both more specific, and workable across zones, is to build input string based on the system default zone offset. That way, the input and output hours should remain unchanged.

Reply:
I guess I do not fully understand why  the `ObjectTimestampFieldConverter` expects the input and output values to be the same if the format pattern has timezone information in it.

Reply:
BTW I do not have it working as I thought I did. Actually the time remains the same. This though does not address the first concern mentioned in the ticket 

> When reading, the Z is accepted in the format but doesn't seem to be taken into account. Indeed, if we have a timestamp with +0200 (plus 2 hours), on output we have the timestamp on the NiFi timezone but without the time offset. For example, with 15h in +0200 I'd have 15h in +0000 instead of 13h in +0000.

Reply:
Please ignore previous comment as I have it working now to be able to change the hour.

Reply:
@exceptionfactory On a separate note, I noticed that `ObjectLocalDateTimeFieldConverter` has special logic for converting when the field is a `Number`.  There are many overloaded `toLocalDateTime` methods to handle the different scenarios. I do not see similar logic in any of the converters for `Number`. Can you please give me insight as to what is the reason this logic is only located in `ObjectLocalDateTimeFieldConverter`?

Reply:
@dan-s1 Yes, that is an important point, as noted in [NIFI-12710](https://issues.apache.org/jira/browse/NIFI-12710) handling `Double` and `Float` support fractional seconds.

Reply:
Thanks for your prompt reply. Why isn't that handled in any of the converters?

Reply:
Since the record `TIMESTAMP` type maps to a `java.sql.Timestamp` the change was focused to particular converter. It may be worth implementing in other converters, but not necessarily. It primarily depends on what provides the conversion for the Record `TIMESTAMP` Type.

Reply:
Refactoring it out may not be necessary at the moment, as other date and time types do not have fractional seconds.

Reply:
I started refactoring it to use it in  `ObjectZonedDateTimeConverter`.  I am trying to clarify if you still have an issue with `ObjectZonedDateTimeConverter` would keep the same behavior as `ObjectLocalDateTimeFieldConverter` if there is no timezone information specified in the format?

Reply:
Thanks for the update. Given that the real focus of the issue is `java.sql.Timestamp`, I recommend building the behavior into `ObjectTimestampFieldConverter` itself. There may be other uses for a `ZonedDateTime` converter in the future, but as we are considering the specialized handling of time zones for `java.sql.Timestamp`, it would be best to implement this in the Timestamp Converter itself.

Reply:
I was able to take most of the tests from `TestObjectLocalDateTimeFieldConverter` and ran them in `ObjectZonedDateTimeConverterTest`.  I replaced `LocalDateTime` with `ZonedDateTime`. The only test I could not get working is testDoubleAsEpochSecondsAsString. To be honest I am not sure what exactly the issue is.

Reply:
@exceptionfactory I see you have been very busy the last couple of days. If you have time can you please review my changes and see if this is what you had in mind? Thanks!

Reply:
Thanks @dan-s1, I plan to get back to this soon.